src/main/java/com/nerodesk/om/aws/AwsDoc.java

```diff
@@ -120,6 +120,11 @@ public void read(@NotNull final OutputStream output) throws IOException {
     @Override
     public void write(final InputStream input, final long size)
         throws IOException {
+        if (size == 0) {
```

Comment:
@longtimeago Let's try to imagine a situation when we need to store a zero size Doc ... I can't think of any. If we will use the decorator SafeDoc we will leave the possibility wrong use of the Doc. I don't like a decorator idea in this case, if we don't allow Null parameter in a method we don't create a decorator for this, right? We check it inside the method.

Even if we still need to store zero-length files then for each implementation is done in its own way. E.g. `File.setLastModified()` for MkDoc and some other way for AwsDoc.